driver/reducecalculation_driver.hpp

```diff
@@ -327,18 +396,24 @@ int ReduceCalculationDriver<Tgpu, Tref>::RunForwardCPU()
         return mloReduceCalculationForwardRunHost<Tgpu, Tref, ReduceCalculationOp_t::Prod>(
             inputDesc, outputDesc, in.data(), outhost.data(), dim, nanPropagation);
     }
+    else if(reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ANY)
+    {
+        return mloReduceLogicalCalculationForwardRunHost<Tgpu, Tref, ReduceCalculationOp_t::Any>(
+            inputDesc, outputDesc, in.data(), outhost.data(), dim);
+    }
 
     return miopenStatusInternalError;
 }
 
-template <typename Tgpu, typename Tref>
-int ReduceCalculationDriver<Tgpu, Tref>::RunBackwardGPU()
+template <typename Tgpu, typename Tref, typename Tgpu_out>
+int ReduceCalculationDriver<Tgpu, Tref, Tgpu_out>::RunBackwardGPU()
 {
-    return miopenStatusSuccess;
+    return miopenStatusNotImplemented;
+    // return miopenStatusSuccess;
 }
 
-template <typename Tgpu, typename Tref>
-Tref ReduceCalculationDriver<Tgpu, Tref>::GetTolerance()
+template <typename Tgpu, typename Tref, typename Tgpu_out>
+Tref ReduceCalculationDriver<Tgpu, Tref, Tgpu_out>::GetTolerance()
```

Comment:
magic number, change to std::numeric_limits<Tgpu>::epsilon() * 10;

Reply:
solved:
```
template <typename Tgpu, typename Tref>
Tref ReduceCalculationDriver<Tgpu, Tref>::GetTolerance()
{
    Tref tolerance = std::numeric_limits<Tgpu>::epsilon() * 10;
    return tolerance;
}
```